test/unit/spec/v2/ion/transformer_spec.js

```diff
@@ -0,0 +1,207 @@
+import transform from 'v2/ion/transformer';
+import XHRFactorRequiredEmail from '../../../helpers/xhr/v2/FACTOR_REQUIRED_EMAIL';
+import XHRFactorVerificationRequiredPush from '../../../helpers/xhr/v2/FACTOR_VERIFICATION_REQUIRED_OKTA_PUSH';
+
+describe('v2/ion/transformer', function () {
+  it('returns result when invokes with invalid resp', () => {
+    expect(transform()).toBeNull();
+    expect(transform('hello')).toBeNull();
+  });
+
+  it('converts factor require email', () => {
+    const result = transform(XHRFactorRequiredEmail.response);
+    expect(result).toEqual({
+      'factor': {
+        'id': 'emf1axecbKovLJPWl0g4',
+        'factorType': 'email',
+        'provider': 'OKTA',
+        'vendorName': 'OKTA',
+        'profile': {
+          'email': 'e...a@rain.com'
+        }
+      },
+      'user': {
+        'id': 'I9bvFiq01cRFgbn',
+        'passwordChanged': '2019-05-03T19:00:00.000Z',
+        'profile': {
+          'login': 'foo@example.com',
+          'firstName': 'Foo',
+          'lastName': 'Bar',
+          'locale': 'en-us',
+          'timeZone': 'UTC'
+        }
+      },
+      'currentState': {
+        'stateHandle': '01OCl7uyAUC4CUqHsObI9bvFiq01cRFgbnpJQ1bz82',
+        'expiresAt': '2018-09-17T23:08:56.000Z',
+        'status': 'FACTOR_REQUIRED',
+        'intent': 'login',
+        'submit-factor': jasmine.any(Function),
+        'cancel': jasmine.any(Function),
+        'context': jasmine.any(Function),
+        'remediation': [
+          {
+            'name': 'submit-factor',
+            'value': [
+              {
+                'name': 'email',
+                'placeholder': 'Enter code',
+                'required': true,
+                'type': 'text'
+              }
+            ]
+          }
+        ]
+      },
+      __raw: XHRFactorRequiredEmail.response,
+    });
+
+    expect(result.currentState['submit-factor']()).toEqual({
+      method: 'POST',
+      url: 'https://your-org.okta.com/api/v2/authn/',
+      data: {
+        stateHandle: '01OCl7uyAUC4CUqHsObI9bvFiq01cRFgbnpJQ1bz82'
+      }
+    });
+
+    expect(result.currentState['submit-factor']({foo: 'bar'})).toEqual({
```

Comment:
Would we ever call the function like this? where we are specifying  the data that needs to be posted? Wouldn't this be available in the response ?

Reply:
sure, will  keep iterate